src/plugins/intel_gpu/src/graph/impls/ocl_v2/col2im.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "col2im.hpp"
+
+#include "col2im_inst.h"
+#include "common_utils/dispatch_utils.hpp"
+#include "intel_gpu/graph/kernel_impl_params.hpp"
+#include "primitive_ocl_base.hpp"
+#include "utils/kernel_generator.hpp"
+
+namespace ov::intel_gpu::ocl {
+namespace {
+
+static std::vector<size_t> get_origin_size(const kernel_impl_params& params) {
+    constexpr size_t spatial_dims = 2;
+    const auto& desc = params.typed_desc<col2im>();
+
+    const auto& output_size = desc->output_shape;
+    const auto& kernel_size = desc->kernel_shape;
+    const auto& stride = desc->stride;
+    const auto& dilation = desc->dilation;
+    const auto& pads_begin = desc->padding_begin;
+    const auto& pads_end = desc->padding_end;
+
+    std::vector<size_t> orig_size(spatial_dims);
+    for (size_t d = 0; d < spatial_dims; ++d) {
+        orig_size[d] = ((output_size[d] + pads_begin[d] + pads_end[d] - (dilation[d] * (kernel_size[d] - 1)) - 1) / stride[d]) + 1;
+    }
+
+    return orig_size;
+}
+
+bool check_ool2im_contain_batch(const kernel_impl_params& params) {
```

Comment:
Did you intend to make `static` function?

Reply:
As we discussed before, col2im only supports static for now.